packages/react/ds/src/radio/radios-group.tsx

```diff
@@ -1,24 +1,34 @@
 'use client';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { ErrorText } from '../error-text/error-text.js';
 import { Heading } from '../heading/heading.js';
 import { HintText } from '../hint-text/hint-text.js';
 import { Radio, getRadioWidth } from './radio.js';
 import type { RadiosGroupType } from './types.js';
 
 export const RadiosGroup = ({
-  fieldId,
+  groupId,
   items,
   inline,
   size,
   errorMessage,
   dividerOption,
   title,
+  onChange,
 }: RadiosGroupType) => {
   const [value, setValue] = useState<null | string>(null);
 
+  useEffect(() => {
+    for (const radio of items) {
+      if (radio.checked) {
+        setValue(radio.value);
+      }
+    }
+  }, []);
+
```

Comment:
If you don't specify all of `useEffect` dependencies I'd suggest adding a comment explaining why you can safely ignore

Reply:
I am not sure why we need this effect? I feel this is a bit confusing..
Could I give the group a list of radios that can have their (default) value checked (say by mistake)? What happens if I do that? Does the last one checked in the list win? Will I see it checked?

Reply:
I would expect the default value of a radio group to be controlled at radio group level, not for each radio?

Reply:
Good point @matteolc . As an alternative, I removed the `useEffect` and also removed the `checked` property from the items array and instead added a `defaultValue` to the group level. If `defaultValue` matches any of the values inside the items array then that will be the default checked radio. Does that sound like a better solution ?